dockerfiles/Dockerfile.build

```diff
@@ -1,17 +1,16 @@
-FROM ubuntu:focal-20220531 AS build
+FROM ghcr.io/edgelesssys/ego/build-base:v0.5.1 AS build
 
-RUN apt update && DEBIAN_FRONTEND=noninteractive apt install -y \
-  build-essential=12.8ubuntu1.1 \
-  clang-10=1:10.0.0-4ubuntu1 \
-  cmake=3.16.3-1ubuntu1 \
+# don't run `apt-get update` because required packages are cached in build-base for reproducibility
```

Comment:
If this is effectively a no-op, I would just remove it? I know it's there so you can quickly switch it back to ubuntu _in theory,_ but then in practice you cannot, given that you don't run `apt-get update` then. That requires some editing, and at this point one could just merge both the build-base and this together anyways.

Reply:
I intentionally only download the packages in build-base. This makes investigating the trustworthiness of build-base easier: compared to the ubunutu base, there are just some .debs added and you can verify their signatures.

Reply:
Ah, did not see the `-d` there. Okay, I guess this makes sense then to use the build-base as a "package repository". I like the idea to use a Docker image here to host all the images, guess that's cleaner then hosting all DEBs separately.